File: qc-core/src/main/java/com/eager/questioncloud/core/domain/point/implement/UserPointReader.java
package com.eager.questioncloud.core.domain.point.implement;

import com.eager.questioncloud.core.domain.point.model.UserPoint;
import com.eager.questioncloud.core.domain.point.repository.UserPointRepository;
import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Component;

@Component
@RequiredArgsConstructor
public class UserPointReader {
    private final UserPointRepository userPointRepository;

    public UserPoint getUserPoint(Long userId) {
        return userPointRepository.getUserPoint(userId);
    }
}

File: qc-core/qc-point/qc-point-api/src/main/kotlin/com/eager/questioncloud/point/implement/UserPointReader.kt
package com.eager.questioncloud.point.implement

import com.eager.questioncloud.point.domain.UserPoint
import com.eager.questioncloud.point.repository.UserPointRepository
import org.springframework.stereotype.Component

@Component
class UserPointReader(
    private val userPointRepository: UserPointRepository,
) {
    fun getUserPoint(userId: Long): UserPoint {
        return userPointRepository.getUserPoint(userId)
    }
}